### DISCUSSION TITLE
MQTT - refreshing values

### DISCUSSION THREAD

**Discussion by Dinth** [john30/ebusd#557] [General]
Hi. Since Ebusd recently introduced HA autodiscovery, im planning to redo whole whole my ebusd setup in HA. Im wondering what is currently the best way of refreshing data coming from Ebusd to HA though? Getting up to date data is very important for me, because for example i control trench radiator fans using that data (switching on the fan when temperature is over 40C and switching them off when it falls down). 
Currently im running a CRON job running a read_all script every 3 minutes and in addition to that every 2 minutes im posting "1" value to /get for each variable i use.

**Comment by Koky05** [comment]
It was discused here:
discussion [#549](https://github.com/john30/ebusd/discussions/549).

**Reply by Dinth** [reply]
Isnt that only related to TCP client? Im asking about MQTT.

**Reply by Koky05** [reply]
You can use automation to /get values you need in interval you expect.
https://github.com/john30/ebusd/wiki/3.3.-MQTT-client

**Reply by Dinth** [reply]
Is there really no "more elegant" way? I find it odd that ebusd requires a manual refresh of the values - not only in one place but in two (readall.sh script + MQTT), not to mention that this is an additional three hundred messages being sent through MQTT server every couple of minutes

**Reply by mwildbolz** [reply]
Maybe the more elegant way would be to let ebusd poll the required data - see #508 (https://github.com/john30/ebusd/issues/508#issuecomment-1022135737)

**Comment by john30** [comment]
if you use the new integration, all matching messages are automatically added to polling due to https://github.com/john30/ebusd/blob/master/contrib/etc/ebusd/mqtt-hassio.cfg#L95-L99
